controllers/control/cassandratask_controller_test.go

```diff
@@ -0,0 +1,278 @@
+package control
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http/httptest"
+	"time"
+
+	"github.com/k8ssandra/cass-operator/pkg/httphelper"
+
+	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	api "github.com/k8ssandra/cass-operator/apis/control/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	mockServer  *httptest.Server
+	callDetails *httphelper.CallDetails
+	// testNamespaceName  = ""
+	testDatacenterName = "dc1"
+)
+
+func createDatacenter(dcName, namespace string) func() {
+	return func() {
```

Comment:
Why does this need to be wrapped in a function?

Reply:
Because of the way ginkgo works `Describe("...", func() ..)` and that's how it was previously used (until I refactored a bit to do it in the beforesuite instead of inside the test)